pkg/apply/desiredset.go

```diff
@@ -36,6 +36,7 @@ type desiredSet struct {
 	informerFactory          InformerFactory
 	remove                   bool
 	noDelete                 bool
+	noDeleteGVK              []schema.GroupVersionKind
```

Comment:
This can be map[schema.GroupVersionKind] so you can do fast lookup and avoid duplicate entry. Overall approaches looks good

Reply:
Ok I will change to that. Add a dummy bool for the value then?

Reply:
yeah dummy struct{} is probably ok.